tsl/test/sql/compression.sql

```diff
@@ -559,9 +570,80 @@ SELECT reloptions FROM pg_class WHERE relname = :statchunk;
 ALTER TABLE stattest SET (autovacuum_enabled = false);
 SELECT decompress_chunk(c) FROM show_chunks('stattest') c;
 SELECT reloptions FROM pg_class WHERE relname = :statchunk;
-
 DROP TABLE stattest;
 
+--- Test that analyze on compression internal table updates stats on original chunks
+CREATE TABLE stattest2(time TIMESTAMPTZ NOT NULL, c1 int, c2 int);
+SELECT create_hypertable('stattest2', 'time', chunk_time_interval=>'1 day'::interval);
+ALTER TABLE stattest2 SET (timescaledb.compress, timescaledb.compress_segmentby='c1');
+INSERT INTO stattest2 SELECT '2020/06/20 01:00'::TIMESTAMPTZ ,1 , generate_series(1, 200, 1);
+INSERT INTO stattest2 SELECT '2020/07/20 01:00'::TIMESTAMPTZ ,1 , generate_series(1, 200, 1);
+
+SELECT  compress_chunk(ch1.schema_name|| '.' || ch1.table_name)
+FROM _timescaledb_catalog.chunk ch1, _timescaledb_catalog.hypertable ht 
+WHERE ch1.hypertable_id = ht.id and ht.table_name like 'stattest2'
+ ORDER BY ch1.id limit 1;
+
+SELECT relname, reltuples, relpages, relallvisible FROM pg_class
+ WHERE relname in ( SELECT ch.table_name FROM 
+                   _timescaledb_catalog.chunk ch, _timescaledb_catalog.hypertable ht
+  WHERE ht.table_name = 'stattest2' AND ch.hypertable_id = ht.id )
+order by relname;
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
+
+--overwrite pg_class stats for the compressed chunk.
+UPDATE pg_class
+SET reltuples = 0, relpages = 0
+ WHERE relname in ( SELECT ch.table_name FROM 
+    _timescaledb_catalog.chunk ch, 
+    _timescaledb_catalog.hypertable ht
+  WHERE ht.table_name = 'stattest2' AND ch.hypertable_id = ht.id 
+        AND ch.compressed_chunk_id > 0 );
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+
+SELECT relname, reltuples, relpages, relallvisible FROM pg_class
+ WHERE relname in ( SELECT ch.table_name FROM 
+                   _timescaledb_catalog.chunk ch, _timescaledb_catalog.hypertable ht
+  WHERE ht.table_name = 'stattest2' AND ch.hypertable_id = ht.id )
+order by relname;
+
+SELECT '_timescaledb_internal.' || compht.table_name as "STAT_COMP_TABLE",
+             compht.table_name  as "STAT_COMP_TABLE_NAME"
+FROM _timescaledb_catalog.hypertable ht, _timescaledb_catalog.hypertable compht
+WHERE ht.table_name = 'stattest2' AND ht.compressed_hypertable_id = compht.id \gset
+
+--analyze the compressed table, will update stats for the raw table.
+ANALYZE :STAT_COMP_TABLE;
```

Comment:
The main feature/test case seems to be missing.

Normally, the user runs ANALYZE on the main hypertable, not the internal compressed table. This is the case we really care about. We want to make sure that ANALYZE on the main hypertable updates the row count for the parent chunks based on the internal compressed chunk.


While it is OK to test also ANALYZE on the internal compressed table, this is not the main case we want to work.

Reply:
@erimatnor We already have a test case for ANALYZE <main table> just above this. Added by another PR that made changes to this area.